src/librustdoc/passes/doc_test_lints.rs

```diff
@@ -73,7 +73,8 @@ pub fn should_have_doc_example(item_kind: &clean::ItemEnum) -> bool {
         | clean::ImportItem(_)
         | clean::PrimitiveItem(_)
         | clean::KeywordItem(_)
-    )
+    ) && (cx.renderinfo.borrow().access_levels.is_public(item.def_id)
+        || cx.render_options.document_private)
```

Comment:
Hmm, I see this used a lot in rustdoc code. Maybe it's worth adding a `cx.appears_in_docs(def_id)` function? Doesn't need to block this PR though.